src/components/signIn/signIn.jsx

```diff
@@ -0,0 +1,78 @@
+import "./signIn.css";
+import { BiDoorOpenFill } from "../../assets/icons/Icons";
+import { Link } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+function SignIn() {
+
+  useEffect(()=>{
+    const timeOut = setTimeout(()=>{
+      setError({msg: "", state: false})
+    },3000)
+    return () => clearTimeout(timeOut);
+  })
+
+  const [error, setError] = useState({ msg: "", state: false });
+  const [loginDetails, setLoginDetails] = useState({
+    userName: "",
+    password: "",
+  });
+
+  const validation = /^(?=.*\d)(?=.*[a-z])([!@#$%]*).{5,}$/;
```

Comment:
suggestion - password validation is not required on signin page, since it would have been already validated during signup.